stumpy/stump_tiles.py

```diff
@@ -0,0 +1,784 @@
+    constant = (m - 1) * m_inverse * m_inverse  # (m - 1)/(m * m)
+    uint64_m = np.uint64(m)
+
+    # for storing diagonal iteration ranges
+    iter_ranges = np.zeros((config.STUMPY_N_DIAGONALS, 3), dtype=np.int64)
+    # for storing covariance values while iterating in chunks
+    covariances = np.zeros(config.STUMPY_N_DIAGONALS, dtype=np.float64)
+    # for storing range of diagonals to traverse in chunks
+    chunk_diags_range = np.arange(config.STUMPY_N_DIAGONALS)
+
+    for tile_idx in range(tiles_ranges[thread_idx, 0], tiles_ranges[thread_idx, 1]):
+        (
+            tile_i,  # i location of current tile in distance profile
+            tile_j,  # j location of current tile in distance profile
+            tile_height,  # height of current tile
+            tile_width,  # width of current tile
+            tile_start_diag,  # index of first diagonal to traverse on current tile
+            tile_stop_diag,  # index of last diagonal to traverse on current tile + 1
+            _,
+        ) = tiles[tile_idx]
+
+        # total number of diagonals to traverse
+        n_diags = tile_stop_diag - tile_start_diag
+        # number of diagonal "chunks" if we divide our diagonals into chunks
+        n_chunks = int(np.ceil(n_diags / config.STUMPY_N_DIAGONALS))
+
+        # chunk_idx is the index of the chunk we are dealing with
+        for chunk_idx in range(0, n_chunks):
+            # number of diagonals to traverse together
+            chunk_size = min(
+                config.STUMPY_N_DIAGONALS,
+                n_diags - (chunk_idx * config.STUMPY_N_DIAGONALS),
+            )
+            # range of diags to traverse in chunk
+            chunk_diags = chunk_diags_range[:chunk_size]
+            # range of diags to traverse in chunk, with respect to entire distance
+            # matrix
+            tile_diags = (
+                tile_start_diag + (chunk_idx * config.STUMPY_N_DIAGONALS) + chunk_diags
+            )
+            # store the diagonal indices we are dealing with, with respect to entire
+            # distance matrix
+            iter_ranges[chunk_diags, 0] = tile_diags - tile_i + tile_j
+            # store the i locations where the diagonals begin
+            iter_ranges[chunk_diags, 1] = tile_i - np.minimum(0, tile_diags)
+            # store the i locations where the diagonals end
+            iter_ranges[chunk_diags, 2] = tile_i + np.minimum(
+                tile_height, tile_width - tile_diags
+            )
+            # record length of longest diagonal
+            longest_diag_len = np.max(
+                iter_ranges[chunk_diags, 2] - iter_ranges[chunk_diags, 1]
+            )
+
+            # row idx is the index of the row of n diagonals we are traversing together
+            for row_idx in range(longest_diag_len):
+                # chunk_diag_idx is the diagonal index we are dealing with,
+                # with respect to the chunk
+                for chunk_diag_idx in range(chunk_size):
```

Comment:
I am curious to understand the point of chunking diagonals for each tile? can't we just traversing diags_of_tile from `tile_start_diag` and tile_stop_diag`? 

I mean...we are not using parallelization here. So, why should we chunk it?